hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java

```diff
@@ -92,6 +100,10 @@
 	private FunctionalityThrottling throttling = null;
 	private DeterministicThrottle.UsageSnapshot gasThrottleUsageSnapshot = NO_GAS_THROTTLE_SNAPSHOT;
 	private DeterministicThrottle.UsageSnapshot[] usageSnapshots = NO_SNAPSHOTS;
+	private long blockNo = 0;
+	private Instant firstConsTimeOfCurrentBlock = NULL_CONSENSUS_TIME;
+	//This initial capacity should prevent the map from rehashing since we will store only 256 pairs and this is less than 0.75*344
+	private Map<Long, Hash> blockNumberToHash = new LinkedHashMap<>(344);
```

Comment:
Since you want to keep this order by key, it would make sense to do,
```suggestion
	private SortedMap<Long, Hash> blockHashes = new TreeMap<>();
```
resizing performance concerns shouldn't matter, since that would happen at most once per 2 seconds.